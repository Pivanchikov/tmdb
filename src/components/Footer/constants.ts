export const linkName = ['about', 'faq']
export const copyright = 'Â© 2021 TMDb, inc.'
