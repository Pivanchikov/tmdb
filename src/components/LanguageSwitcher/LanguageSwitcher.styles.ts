export const styles = theme => ({
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    borderRadius: 4,
    boxShadow: theme.shadows[5],
    padding: theme.spacing(4),
    width: 360,
  },
  languageSwitcher: {
    minWidth: 28,
    height: 26,
    border: `1px solid ${theme.palette.secondary.main}`,
    borderRadius: 3,
    padding: 1,
    fontSize: '0.8rem',
    alignItems: 'center',
    justifyContent: 'center',
    display: 'flex',
    backgroundColor: theme.palette.primary.main,
    cursor: 'pointer',
    color: theme.palette.secondary.main,
    '&:hover': {
      backgroundColor: theme.palette.secondary.main,
      color: theme.palette.primary.main,
    },
  },
  label: {
    color: 'inherit',
  },
  closeIcon: {
    fill: theme.palette.primary.main,
    cursor: 'pointer',
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  title: {
    color: theme.palette.primary.main,
  },
  item: {
    color: theme.palette.primary.main,
    width: 150,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  marginRight: {
    marginRight: 10,
  },
  container: {
    display: 'flex',
    marginTop: 20,
    justifyContent: 'space-around',
  },
})
