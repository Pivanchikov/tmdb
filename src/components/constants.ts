export const films = ['popular', 'nowPlaying', 'upcoming', 'topRated']
export const serials = ['popular', 'airingToday', 'onTv', 'topRated']
export const people = ['popularPeople']
export const another = ['support', 'API']
export const burgerItems = {
  top: ['movies', 'tvShows', 'people'],
  middle: ['API', 'support', 'about'],
  bottom: ['signIn', 'signUp'],
}
export const languages = [
  ['Ru', 'ru'],
  ['Gb', 'en'],
]
export const desktopWidth = 1280
export const TIMEOUT_MS = 300
export const QUERY_LENGTH_THRESHOLD = 3
