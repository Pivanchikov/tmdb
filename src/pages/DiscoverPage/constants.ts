export const STEP = 1
export const MIN_YEAR = 1920
export const MAX_YEAR = 2021
