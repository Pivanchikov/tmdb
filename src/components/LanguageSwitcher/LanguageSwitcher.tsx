import { useState, useRef } from 'react'
import { MenuItem, makeStyles, Button, Modal, Backdrop, Typography } from '@material-ui/core'
import { languages } from '../constants'
import { useTranslation } from 'react-i18next'
import ReactCountryFlag from 'react-country-flag'
import CloseIcon from '@material-ui/icons/Close'

import { styles } from './LanguageSwitcher.styles'

const useStyles = makeStyles(styles)

export const LanguageSwitcher = () => {
  const classes = useStyles()
  const [opened, setOpened] = useState(false)
  const { t, i18n } = useTranslation()

  const anchorRef = useRef<HTMLButtonElement>(null)

  const handleToggle = () => setOpened(state => !state)
  const closeMenu = () => setOpened(false)
  const changeLanguage = e => {
    i18n.changeLanguage(e.target.id)
    closeMenu()
  }

  return (
    <>
      <Button
        ref={anchorRef}
        onClick={handleToggle}
        classes={{ root: classes.languageSwitcher, label: classes.label }}
      >
        {i18n.language}
      </Button>
      <Modal
        className={classes.modal}
        open={opened}
        onClose={handleToggle}
        BackdropComponent={Backdrop}
        BackdropProps={{
          timeout: 500,
        }}
      >
        <div className={classes.paper}>
          <header className={classes.header}>
            <Typography variant="h5" className={classes.title}>
              {t('selectLanguage')}
            </Typography>
            <CloseIcon
              fontSize="medium"
              classes={{ root: classes.closeIcon }}
              onClick={handleToggle}
              data-testid="closeModal"
            />
          </header>
          <section className={classes.container}>
            {languages.map((item, index) => {
              const [flag, languageCode] = item

              return (
                <MenuItem
                  className={classes.item}
                  key={index}
                  id={languageCode}
                  onClick={changeLanguage}
                  data-testid="lang"
                >
                  <ReactCountryFlag countryCode={flag} svg className={classes.marginRight} />
                  {t(languageCode)}
                </MenuItem>
              )
            })}
          </section>
        </div>
      </Modal>
    </>
  )
}
