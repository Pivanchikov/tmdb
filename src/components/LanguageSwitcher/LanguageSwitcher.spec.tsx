import { render as testingLibraryRender, fireEvent } from '@testing-library/react'

import { LanguageSwitcher } from './LanguageSwitcher'
import { languages } from '../constants'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

describe('LanguageSwitcher', () => {
  const render = () => testingLibraryRender(<LanguageSwitcher />)

  it('renders correctly initial document', () => {
    const { getByRole, getAllByTestId } = render()

    fireEvent.click(getByRole('button'))

    expect(getAllByTestId('lang')).toHaveLength(languages.length)
  })

  it('changing language', () => {
    const { getByRole, getAllByTestId } = render()

    fireEvent.click(getByRole('button'))
    const [item] = getAllByTestId('lang')

    fireEvent.click(item)
    expect(jest.fn()).toHaveBeenCalled
  })

  it('matches snapshot', () => {
    const { asFragment } = render()

    expect(asFragment()).toMatchSnapshot()
  })
})
