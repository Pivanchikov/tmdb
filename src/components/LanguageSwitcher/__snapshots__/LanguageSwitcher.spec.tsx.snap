// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root makeStyles-languageSwitcher-23 MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label makeStyles-label-24"
    >
      en
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
